### PR TITLE
ITrace: Fixes concurrency of cache. 

### PR DESCRIPTION
**Pull request by bchong95**
# ITrace: Fixes concurrency of cache.

## Description

Makes the cache concurrent.

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Tracing/TraceWriter.cs [hunk 1]
```diff
@@ -4,15 +4,15 @@
...
+    using System.Collections.Concurrent;
     using System.IO;
     using System.Linq;
     using System.Text;
     using Microsoft.Azure.Cosmos.Json;
 
     internal static partial class TraceWriter
     {
-        private static readonly Dictionary<string, string> FilePathToName = new Dictionary<string, string>();
+        private static readonly ConcurrentDictionary<string, string> FilePathToName = new ConcurrentDictionary<string, string>();
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Tracing/TraceWriter.cs:15]: Please add tests.

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/Tracing/TraceWriter.cs:15]: Let's avoid regressing this, please add a test using concurrent tasks.

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/Tracing/TraceWriter.cs:15]: A test would also discover any other thread-unsafe operations that could be happening.

**Review by sboshra — APPROVED** [review]
:shipit:

**Pull request merged**